lib/postgrex/utils.ex

```diff
@@ -47,15 +47,23 @@ defmodule Postgrex.Utils do
   Fills in the given `opts` with default options.
   """
   @spec default_opts(Keyword.t) :: Keyword.t
-  def default_opts(opts) do
+  def default_opts(opts, environment \\ System) do
     opts
-    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
-    |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
-    |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
-    |> Keyword.put_new(:port, System.get_env("PGPORT"))
+    |> Keyword.put_new(:username, environment.get_env("PGUSER") || environment.get_env("USER"))
+    |> Keyword.put_new(:password, environment.get_env("PGPASSWORD"))
+    |> Keyword.put_new(:hostname, environment.get_env("PGHOST") || "localhost")
+    |> Keyword.put_new(:port, set_port(environment.get_env("PGPORT")))
```

Comment:
So, I'm trying to evaluate what to do instead of this. I have a few issues:

* Using `after` I can't grab the previous set variable inside the function it seems, like this:
```elixir
  test "translates port to integer" do
    opts = []
    previous_port = System.get_env("PGPORT")
    assert 5432 == U.default_opts(opts, Environment)[:port]
  after
    set_back_pgport(previous_port)
  end
```

I'm getting a `undefined function previous_port/0` error.

If I don't use an `after` block, I'm assuming it might not get set back if the test fails based on your reasoning on why to use `after`.

One other concern is that env vars being directly set will possibly leak onto other tests as they run concurrently, won't they?